coalib/bears/meta.py

```diff
@@ -0,0 +1,25 @@
+from collections import defaultdict
+
+
+class bearclass(type):
+    """
+    Metaclass for :class:`coalib.bears.Bear.Bear` and therefore all bear
+    classes.
+
+    Pushing bears into the future... ;)
+    """
+
+    # by default a bear class has no aspects
```

Comment:
I would like rephrase your last line "this bear can handle ALL possible aspects it can". With the help of aspects we can restrict the analysis of a bear. We can force it to analyse only a particular aspect, but we never can do an analysis the bear cannot do because bear is just wrapping the tool. So my point was if by default the aspect was root aspect, it can be beneficiary at the point when bears are called explicitly. We will divide bears into segments and a segment is run only when we pass that aspect and as Root aspect is parent aspect every aspect the segment will run even when no aspects are passed.